### PR TITLE
Add NumpyConversionOperator

### PR DESCRIPTION
**Pull request by sdrave**
Also use `NumpyConversionOperator` in the discretizers to ensure that `output_functional.range` always is a `NumpyVectorSpace`.

### CODE REVIEW

File: src/pymor/operators/constructions.py [hunk 1]
```diff
@@ -1362,3 +1362,55 @@ def __call__(self, U, mu=None):
...
+
+    def __init__(self, space, direction='to_numpy'):
+        assert direction in ('to_numpy', 'from_numpy')
+        self.__auto_init(locals())
+        if direction == 'to_numpy':
+            self.source = space
+            self.range = NumpyVectorSpace(space.dim, id=space.id)
+        else:
+            self.range = space
+            self.source = NumpyVectorSpace(space.dim, id=space.id)
```
Comment (TiKeil) [inline src/pymor/operators/constructions.py:1396]:
```suggestion
            self.source = NumpyVectorSpace(space.dim, id=space.id)
            self.range = space
```
I think this is a better way to sort this for readability.

File: src/pymor/operators/constructions.py [hunk 2]
```diff
@@ -1362,3 +1362,55 @@ def __call__(self, U, mu=None):
         if self.raise_negative and np.any(norm_squared < 0):
             raise ValueError(f'norm is negative (square = {norm_squared})')
         return np.sqrt(norm_squared)
+
+
+class NumpyConversionOperator(Operator):
+    """Converts |VectorArrays| to |NumpyVectorArrays|.
+
+    Note that the input |VectorArrays| need to support
+    :meth:`~pymor.vectorarrays.interface.from_numpy`.
```
Comment (TiKeil) [inline src/pymor/operators/constructions.py:1371]: Can it happen that `from_numpy` exists, but `to_numpy` does not?

**Review by TiKeil — CHANGES_REQUESTED** [review]
This is a very nice idea. The issue with these `BlockColumnOperator`s for the vector valued actually also bothered me already and is a nice way to make `estimate_image` work for #1474

File: src/pymor/operators/constructions.py (hunk 2 above)
Comment (sdrave) [reply src/pymor/operators/constructions.py:1371]: Yes. Neither method is guaranteed to work for an arbitrary `VectorArray`. The operator should only be used in 'safe' situations.

Comment (TiKeil) [reply src/pymor/operators/constructions.py:1371]: so here one would have to add something like `or :meth:`~pymor.vectorarrays.interface.to_numpy, respectively` .

**Review by TiKeil — APPROVED** [review]

**Comment by sdrave** [comment]
@TiKeil I have pushed a few more commits, which fix `output_d_mu` for complex outputs (there was a warning in the test log that `array[index] = ...` truncates the imaginary part) and implement `d_mu` for `ConcatenationOperator`. Can you take another look?

**Comment by TiKeil** [comment]
> @TiKeil I have pushed a few more commits, which fix `output_d_mu` for complex outputs (there was a warning in the test log that `array[index] = ...` truncates the imaginary part) and implement `d_mu` for `ConcatenationOperator`. Can you take another look?

Ah yes, thanks for fixing the warning. Totally makes sense. 
I totally agree with the product rule for `ConcatenationOperator`.

**Pull request merged**